azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -884,7 +884,7 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 		}
 
 		// changing rbacEnabled must still force cluster recreation
-		if *props.EnableRBAC == rbacEnabled {
+		if *props.EnableRBAC == rbacEnabled && !*props.AadProfile.Managed {
```

Comment:
There is a potential crash here so we'll want to nil check this before pointing to it

Reply:
Tried to resolve in a good way, but my Go experience is too limited to know what the best way is